classes/local/store/object_file_system.php

```diff
@@ -756,6 +762,29 @@ public function generate_presigned_url_to_external_file($contenthash, $headers =
         return $this->externalclient->generate_presigned_url($contenthash, $headers);
     }
 
+    /**
+     * Check if file extension is whitelisted.
+     * @param string $filename
+     * @return bool
+     * @throws \dml_exception
+     */
+    public function is_extension_whitelisted($filename) {
+        $config = manager::get_objectfs_config();
+        if (empty($config->signingwhitelist)) {
+            return false;
+        }
+        $whitelist = explode(',',  $config->signingwhitelist);
+        if (empty($whitelist)) {
+            return false;
+        }
+        $extension = pathinfo($filename, PATHINFO_EXTENSION);
```

Comment:
It would be better to not implement this logic, there should be a core helper for this.

Look at other code which uses the admin_setting_filetypes, also talk to peter as he did work on this in another plugin which is very similar:

https://tracker.moodle.org/browse/MDL-54716

https://github.com/moodle/moodle/compare/master...Peterburnett:MDL-54716#diff-f2b5c8f463bfa8a6613933d9b305cf57R784-R785

     $util = new core_form\filetypes_util();
        $sitewidetypes = $util->normalize_file_types($CFG->allowedfiletypes);